packages/tupaia-web/src/features/Map/MapOverlays/PolygonNavigationLayer.tsx

```diff
@@ -63,20 +62,23 @@ export const PolygonNavigationLayer = () => {
   const { projectCode, entityCode } = useParams();
   const { selectedOverlay } = useMapOverlays(projectCode, entityCode);
   const { data: activeEntity } = useEntity(projectCode, entityCode);
+
   const { data: entities } = useEntitiesWithLocation(projectCode, entityCode);
 
   if (!entities || entities.length === 0) {
     return null;
   }
 
   const childEntities = entities.filter((entity: Entity) => entity.parentCode === entityCode);
-
   const showChildEntities =
-    !POLYGON_MEASURE_TYPES.includes(selectedOverlay?.displayType) && childEntities?.length > 0;
+    POLYGON_MEASURE_TYPES.includes(selectedOverlay?.displayType) && childEntities?.length > 0;
```

Comment:
I changed this as it seemed like the reverse of what the current web frontend is doing

Reply:
hmmm I'm not sure about this. It's quite hard to follow in the OG codebase but what this is meant to do in this version is only show child polygons if the selected map overlay is not a polygon type because then there would be polygons displayed on top of polygons. Also if there is no selected overlay this should show the children but after the change it doesn't. Is this fixing a bug that was found in testing?